android/java/org/chromium/chrome/browser/vpn/activities/BraveVpnParentActivity.java

```diff
@@ -210,6 +229,18 @@ public void onGetWireguardProfileCredentials(
         }
     }
 
+    public void changeServerRegion() {
+        mIsServerLocationChanged = true;
+        BraveVpnUtils.showProgressDialog(
+                BraveVpnParentActivity.this, getResources().getString(R.string.vpn_connect_text));
+        if (BraveVpnNativeWorker.getInstance().isPurchasedUser()) {
+            mBraveVpnPrefModel = new BraveVpnPrefModel();
+            BraveVpnNativeWorker.getInstance().getSubscriberCredentialV12();
```

Comment:
not related to PR, but a bit weird that getter function doesn't return anything

Reply:
@samartnik it's a function to get subscriber credential from guardian API. and we have the callback with the response.

Reply:
I had trouble thinking about this too @samartnik - but this class itself is an observer (see https://github.com/brave/brave-core/blob/689a59643a73ee73cbe19e095122e35ff67f469f/android/java/org/chromium/chrome/browser/vpn/BraveVpnObserver.java#L29) and the implementation here gets called after response from Guardian.

Reply:
@bsclifton it's an api call so we have to get the response in the callback. and with jni call, we have to do it through observer. do we have any concerns here ?

Reply:
we can get rid of it once we migrate to mojom interface. i have introduced it with new api in the PR. we need to refactor the changes when we can.

Reply:
No concerns! and yep- mojom right way to handle 😄 Was just sharing to @samartnik that it took me a minute to parse in my head (noticing observer, how JNI is registered, etc) when tracing. Thanks!